datafusion/expr/src/udaf.rs

```diff
@@ -249,6 +249,14 @@ impl AggregateUDF {
     pub fn simplify(&self) -> Option<AggregateFunctionSimplification> {
         self.inner.simplify()
     }
+
+    /// Returns true if the function is max, false if the function is min
+    /// None in all other cases, used in certain optimizations or
+    /// or aggregate
+    ///
+    pub fn get_minmax_desc(&self) -> Option<bool> {
```

Comment:
Can I suggest using an `enum` like `enum MinMax { Min, Max }` instead of a `bool`?

Reply:
Here is the code that we need the boolean, maybe name it with `is_descending: Option<bool>`?
```rust
        let (field, desc) = aggr.get_minmax_desc()?;
        if desc != order.options.descending {
            return None;
        }
```

Reply:
Fixed with [`eb0ebe0` (#11696)](https://github.com/apache/datafusion/pull/11696/commits/eb0ebe04fc92baec527ffd945b8801f72ffecf48) although I am not sure I understood what "is descending" mean for a function?

Reply:
It is the same idea like descending in `SortOptions`

```rust
pub struct SortOptions {
    /// Whether to sort in descending order
    pub descending: bool,
```

Would `is_sorted_descending: Option<bool>` be much more better?

Reply:
let field = aggr.field()
let is_sorted_descending = aggr.is_sorted_descending()

And, I think getting field and descending separately is better than `get_minmax_desc`, so if we can get `field` independently if we don;t care about descending

Reply:
Thanks. The goal of this PR was to simplify #10943 by implementing here the necessary method for udaf, do you think I should proceed in the refactoring now or can we do as a separate improvement?

Reply:
Not necessary